ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,389 @@
+            warnings.warn(msg)
+
+    def _prepare_subarray_info(self, instrument_description):
+        """
+        Constructs a SubArrayDescription object from
+        self.file_.root.configuration.instrument.telescope.optics
+        and
+        self.file_.root.configuration.instrument.subarray.layout
+        tables.
+
+        Returns
+        -------
+        SubarrayDescription :
+            instrumental information
+        """
+        available_optics = instrument_description.telescope.optics.iterrows()
+        available_telescopes = instrument_description.subarray.layout.iterrows()
+
+        # The focal length choice is missing here
+        # I am not sure how they are handled in the file
+        # Will there be one of "e_fl" or "fl" in the columns?
+        # This will only work if "e_fl" is available
+        optic_descriptions = {}
+        for optic in available_optics:
+            optic_description = OpticsDescription(
+                name=optic['name'].decode(),
+                num_mirrors=optic['num_mirrors'],
+                equivalent_focal_length=u.Quantity(
+                    optic['equivalent_focal_length'],
+                    u.m,
+                ),
+                mirror_area=u.Quantity(
+                    optic['mirror_area'],
+                    u.m ** 2,
+                ),
+                num_mirror_tiles=optic['num_mirror_tiles'],
+            )
+            optic_descriptions[optic['description'].decode()] = optic_description
+
+        tel_positions = {}
+        tel_descriptions = {}
+        for telescope in available_telescopes:
+            tel_positions[telescope['tel_id']] = (
+                telescope['pos_x'],
+                telescope['pos_y'],
+                telescope['pos_z'],
+            )
+            geom = CameraGeometry.from_name(telescope['camera_type'].decode())
+            optics = optic_descriptions[telescope['tel_description'].decode()]
+            tel_descriptions[telescope['tel_id']] = TelescopeDescription(
+                name=telescope['name'],
+                tel_type=telescope['type'],
+                optics=optics,
+                camera=geom,
+            )
+
+        return SubarrayDescription(
+            name='???',
+            tel_positions=tel_positions,
+            tel_descriptions=tel_descriptions,
+        )
+
+    def _parse_mc_header(self, header_table):
+        """
+        Construct a MCHeaderContainer from the
+        self.file_.root.configuration.simulation.run
+        """
+        # We just assume there is only one row?
+        # Thats kind of what the SimTelEventSource does
+        # If mixed configurations are in one file, it gets problematic anyway, right? -> subarray layout?
+        header = header_table.row
+        return MCHeaderContainer(
+            corsika_version=header["corsika_version"],
+            simtel_version=header["simtel_version"],
+            energy_range_min=header["energy_range_min"] * u.TeV,
```

Comment:
always use `u.Quantity(value, unit)` instead of multiplication

Reply:
If value is an array, also add `copy=False`

Reply:
also you don't need to do this at all.  There is a `ctapipe.io.TableReader` that turns a table back into the relevant objects, row by row.  It  should not be done manually.  see:
*  https://cta-observatory.github.io/ctapipe/api/ctapipe.io.HDF5TableReader.html?highlight=hdf5tablereader#ctapipe.io.HDF5TableReader 
* or the tutorial here:
https://cta-observatory.github.io/ctapipe/examples/table_writer_reader.html#Reading-one-row-at-a-time: